x-pack/plugins/ml/public/application/jobs/new_job/pages/components/pick_fields_step/components/population_view/metric_selection.tsx

```diff
@@ -36,7 +37,7 @@ export const PopulationDetectors: FC<Props> = ({ setIsValid }) => {
   } = useContext(JobCreatorContext);
   const jobCreator = jc as PopulationJobCreator;
 
-  const { fields } = newJobCapsService;
+  const [fields] = useState(sortFields([...newJobCapsService.fields, ...jobCreator.runtimeFields]));
```

Comment:
Does this need `useState` if a `setFields` is never needed? Maybe it doesn't need to be wrapped in `useState` at all or can make use of `useMemo`?

Reply:
updated in https://github.com/elastic/kibana/pull/91168/commits/3e19ad9587a3e70040dcf4523c0e377a0bd54ae2